### DISCUSSION TITLE
Android: App incorrectly handles "Login with GitHub" links

### DISCUSSION THREAD

**Discussion by fsimonis** [community/community#3489] [Mobile] [answered]
##  Problem Description
The GitHub app (v 1.4.11) currently opens all `github.com` links.
This includes links for "Login with GitHub" on third party pages.
The GitHub app itself then opens **a** browser with the login page.

##  Expected Behaviour
The GitHub App doesn't handle the link and lets the browser handle the login page.
The GitHub App open the login page in the default browser.

## Example
This leads to the following problem on my Samsung phone:
* Using Firefox (v 86.1.1) with "open links in apps" enabled and set as default browser
* Click on a "login with GitHub" e.g. [gitter.im](https://gitter.im/login/#login)
* The GitHub App opens
* The Samsung Internet Browser opens showing the login page, which is pointless as I am using Firefox and the login doesn't _transfer_ to Firefox.

A fix for this issue is to temporarily disable "open links in apps" in Firefox mobile before signing into GitHub.

**Comment by aconsuegra** [comment]
Hey @fsimonis. Thanks for your report! 

We just released a new Beta in Google Play (v 1.7.1) with a potential fix for this issue. You are more than welcome to join our Beta program to help us validate the fix, but if you'd rather wait for a production build, this version should be publicly released next week.

Thanks

**Reply by 5678bgu** [reply]
[]()

**Reply by aurelienheude** [reply]
super test

**Comment by PranamBhat** [comment, marked as answer]
Kindly turn on "Automatic Update" > "ON" for GitHub mobile application to get the latest update once they release.

**Reply by fsimonis** [reply]
The update fixed the problem.
Thank you!

**Reply by aurelienheude** [reply]
true

**Comment by snarfed** [comment]
Sadly it looks like this regressed recently. When I try to log into github.com in a browser on Android, it redirects to the app and says "Something went wrong" with a "Try again" button.

I'm on a Pixel 7a, Android 13, GitHub app 1.121.1. Happens in both Chrome and Firefox.

**Reply by MustafaAvci0q** [reply]
No

**Reply by MustafaAvci0q** [reply]
> Sadly it looks like this regressed recently. When I try to log into github.com in a browser on Android, it redirects to the app and says "Something went wrong" with a "Try again" button.
> 
> I'm on a Pixel 7a, Android 13, GitHub app 1.121.1. Happens in both Chrome and Firefox.

**Reply by Omegastick** [reply]
Confirmed on a Samsung Galaxy S22 and GitHub app version 1.126.0.

**Reply by aconsuegra** [reply]
Hi, we are aware of the regression that caused some issues recently supporting login into github.com. Tomorrow we are releasing a new beta version that will include a potential fix, so we would be pleased if you could give it a try to verify the behavior. Thanks in advance!

**Reply by aurelienheude** [reply]
mmmh

**Reply by snarfed** [reply]
@aconsuegra apologies for the delay here! The potential fix evidently didn't work for me, sorry, I still hit this bug regularly on v1.146.0. Let me know if I can help debug!

**Reply by ByteSizedMarius** [reply]
Same issue for me in 1.151.0.

**Reply by maciejmatczak** [reply]
Android, Firefox, still "Something went wrong" when app handles login redirection from browser.

**Comment by MustafaAvci0q** [comment]
> Kindly turn on "Automatic Update" > "ON" for GitHub mobile application to get the latest update once they release.

**Comment by Tolgahan0** [comment]
> ##
Introduction:
GitHub's mobile app has been a go-to platform for developers to access repositories, collaborate on projects, and stay updated with the latest developments. However, a recent issue has emerged where the GitHub app opens all github.com links, including "Login with GitHub" links on third-party pages. This can be frustrating for users, especially when the app opens a browser for the login page. In this post, we'll discuss the impact of this issue and propose a potential solution to enhance the overall user experience.

Understanding the Issue:
The current version of the GitHub app (v 1.4.11) automatically opens all github.com links, even when those links are intended for third-party login integration. This behavior can lead to confusion for users who expect a more integrated experience. The "Login with GitHub" feature, widely used by many third-party services, loses its convenience when the GitHub app directs users to a separate browser session for authentication.

Proposed Solution:
To address this issue and provide a smoother user experience, GitHub could consider implementing a solution that distinguishes between direct repository links and third-party login links. Here's how the proposed solution might work:

Whitelisting Domains:
GitHub's mobile app could be updated to whitelist specific domains for third-party services that use "Login with GitHub." This means that links from these whitelisted domains would still be opened in the GitHub app, offering a consistent and seamless experience.

Custom Link Handling:
For non-whitelisted domains, the app could handle the links differently. Instead of automatically opening a browser, the app could display a dialogue box or notification when users click on such links. This dialogue could explain that the link is intended for third-party login and provide the option to open the link in an external browser if the user prefers.

User Preference:
To give users more control over their experience, the GitHub app could include a setting that allows users to choose whether they want github.com links to open within the app or in an external browser. This way, users who frequently use third-party integrations could opt to have those links opened externally, while still enjoying the integrated experience for repository links.

Conclusion:
By implementing these solutions, GitHub can enhance the user experience on its mobile app by intelligently handling different types of github.com links. This approach not only maintains the app's seamless navigation for repository-related links but also ensures that third-party login links are appropriately handled. This change would demonstrate GitHub's commitment to providing a cohesive and user-centric experience for developers and users alike, addressing the current issue and fostering a more integrated mobile experience.

**Reply by 5678bgu** [reply]
[]()

**Comment by phuquoc81** [comment]
App add features section coding of game developer

**Comment by snarfed** [comment]
Status update! This is working ok now in Chrome (126.0.6478.110), but it's still broken in Firefox (127.0.1). GitHub app 1.165.0, Pixel 7a, Android 14.

**Comment by alexmdev04** [comment]
Currently broken again, Brave 127.0.6533.100, GitHub App 1.173.0, Android 12, POCO X3 NFC MIUI Global 14.0.5.

When clicking sign in on the GitHub login website in Brave it gives a prompt to choose an app to open the link in - it lists all browsers installed, but selecting a browser does nothing and the GitHub website does not get redirected, it just says "Signing In...". A 2FA request does come through but you can't confirm the digits because the browser never redirected (the 2FA request with no digits required also does nothing). 

The fix for MIUI is to use another app (like nova launcher) to open a shortcut to the vanilla android app management page, select GitHub, select open by default, disable open supported links.

**Comment by sheilbronn** [comment]
The login on (Android) Github Mobile 1.202.0 seems to be broken (again).
When clicking on "_Login to Github.com_" (mine is German: "Bei Github.com anmelden")
there is a popup saying s.th. like:
_Login not possible. It is recommended to install a browser such as Chrome or Firefox._
(In my case in German: "Anmeldung nicht möglich. Es wird empfohlen, einen Browser wie Chrome oder Fiefox zu installieren.")
I've tried it with Chrome, Firefox and Samsung Internet as default browsers - with no success.
Login from the web page in Firefox works fine.

**Comment by abdohelalnada123456-pixel** [comment]
Can you explain the laws?

**Reply by viogimbul3-maker** [reply]
Dapatkan vertifikasi kode wathsapp saya
